### PR TITLE
[Fix_#1989] Considering '.' in variable name when merging variables

### LINKED ISSUE #1989: Duplicated information in the `variables` object
### Describe the bug

Sometimes when `variables` are fetched from the Data Index, there is duplicated information in the response.

### Expected behavior

I'd expect simply:
```json
"variables": {
  "workflowdata": {
    "myKey": "myValue",
    "mantra": "Serverless Workflow is awesome!",
    "greeting": "Hello World"
  }
}
```

### Actual behavior

If you fetch the `variables`, you'll get:
```json
"variables": {
  "workflowdata": {
    "myKey": "myValue",
    "mantra": "Serverless Workflow is awesome!",
    "greeting": "Hello World"
  },
  "workflowdata.mantra": "Serverless Workflow is awesome!",
  "workflowdata.greeting": "Hello World"
}
```

See that `mantra` and `greeting` are duplicated.

### How to Reproduce?

Take the [Hello World workflow](https://github.com/apache/incubator-kie-kogito-examples/blob/stable/serverless-workflow-examples/serverless-workflow-hello-world/src/main/resources/hello.sw.json) as an example and execute it with additional inputs, for instance:
```json
{
    "myKey": "myValue"
}
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Kogito version or git rev (or at least Quarkus version if you are using Kogito via Quarkus platform BOM)

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by fjtirado**
Fixes https://github.com/apache/incubator-kie-kogito-apps/issues/1989

### CODE REVIEW

File: data-index/data-index-common/src/main/java/org/kie/kogito/index/json/JsonUtils.java [hunk 1]
```diff
@@ -50,4 +54,20 @@ public static ObjectMapper configure(ObjectMapper objectMapper) {
         objectMapper.registerModule(module);
         return objectMapper;
     }
+
+    public static ObjectNode mergeVariable(String variableName, Object variableValue, ObjectNode variables) {
+        return (ObjectNode) MergeUtils.merge(getObjectNode(variableName, variableValue), variables);
+    }
+
+    private static ObjectNode getObjectNode(String variableName, Object variableValue) {
```
Comment (wmedvede) [inline data-index/data-index-common/src/main/java/org/kie/kogito/index/json/JsonUtils.java]: Sorry for being picky, but i believe this method name with "get" is a bit confusing, Id prefer a name like createVariableNode, or something similar.

**Review by nmirasch — APPROVED** [review]

File: data-index/data-index-common/src/test/java/org/kie/kogito/index/json/JsonUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Test
+    void testSimpleMerge() {
+        ObjectNode expected = ObjectMapperFactory.get().createObjectNode().put("Javierito", "manolo");
+        assertThat(JsonUtils.mergeVariable("Javierito", "manolo", null)).isEqualTo(expected);
+    }
+
+    @Test
+    void testComplexMergeWithDot() {
+        ObjectNode expected = ObjectMapperFactory.get().createObjectNode().set("Javier", ObjectMapperFactory.get().createObjectNode().put("ito", "manolo").put("ato", "pepe"));
```
Comment (wmedvede) [inline data-index/data-index-common/src/test/java/org/kie/kogito/index/json/JsonUtilsTest.java]: your call, but I believe using more standard names, "bob", "alice", "user1", "user2" are more follow-able.

**Review by wmedvede — APPROVED** [review]
Minor comments from my side.

File: data-index/data-index-common/src/main/java/org/kie/kogito/index/json/JsonUtils.java (hunk 1 above)
Comment (fjtirado) [reply data-index/data-index-common/src/main/java/org/kie/kogito/index/json/JsonUtils.java]: yes, you are right

File: data-index/data-index-common/src/test/java/org/kie/kogito/index/json/JsonUtilsTest.java (hunk 2 above)
Comment (fjtirado) [reply data-index/data-index-common/src/test/java/org/kie/kogito/index/json/JsonUtilsTest.java]: I have to stick with Manolo and Pepe for patriotic reasons ;)
Changing Javier.ito to key.user1 ;)

**Pull request merged**